From 46c99e3d7cf38446491b3c2a826fc05dcebc588d Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Tue, 16 Nov 2021 12:07:29 +0000
Subject: [PATCH 1249/2201] Revert "overlays: vc4-kms-v3d: Change composite
 handling"

This reverts commit 063a326cbaff498c7cd001648b400f12c0ec4237.

Reverting temporarily because not enabling the composite output has
been seen to break HDMI on older Pis.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/README                  | 4 ++--
 arch/arm/boot/dts/overlays/upstream-overlay.dts    | 2 +-
 arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts | 4 ++--
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index bedb501b3ce5..8e4939377c40 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -3620,8 +3620,8 @@ Params: cma-512                 CMA is 512MB (needs 1GB)
         cma-default             Use upstream's default value
         audio                   Enable or disable audio over HDMI (default "on")
         noaudio                 Disable all HDMI audio (default "off")
-        composite               Enable the composite output (default "off")
-                                N.B. Disables all other outputs on a Pi 4.
+        nocomposite             Disable the composite video output (default
+                                "off")
 
 
 Name:   vc4-kms-v3d-pi4
diff --git a/arch/arm/boot/dts/overlays/upstream-overlay.dts b/arch/arm/boot/dts/overlays/upstream-overlay.dts
index 2852bea52309..7c4071a7cb27 100644
--- a/arch/arm/boot/dts/overlays/upstream-overlay.dts
+++ b/arch/arm/boot/dts/overlays/upstream-overlay.dts
@@ -1,4 +1,4 @@
-// redo: ovmerge -c vc4-kms-v3d-overlay.dts,cma-default,composite dwc2-overlay.dts,dr_mode=otg
+// redo: ovmerge -c vc4-kms-v3d-overlay.dts,cma-default dwc2-overlay.dts,dr_mode=otg
 
 /dts-v1/;
 /plugin/;
diff --git a/arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts b/arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts
index 351fc160e803..62e1d77a8182 100644
--- a/arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts
+++ b/arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts
@@ -89,7 +89,7 @@
 
 	fragment@11 {
 		target = <&vec>;
-		__dormant__  {
+		__overlay__  {
 			status = "okay";
 		};
 	};
@@ -118,6 +118,6 @@
 	__overrides__ {
 		audio   = <0>,"!13";
 		noaudio = <0>,"=13";
-		composite = <0>, "=11";
+		nocomposite = <0>, "!11";
 	};
 };
-- 
2.17.1

